File: app/src/main/java/ostmodern/skylark/repository/local/SkylarkDatabase.java
package ostmodern.skylark.repository.local;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.RoomDatabase;

@Database(entities = {SetEntity.class, EpisodeEntity.class, FavouriteEntity.class}, version = 1, exportSchema = false)
public abstract class SkylarkDatabase extends RoomDatabase {
    public abstract SetDao setDao();

    public abstract EpisodeDao episodeDao();

    public abstract FavouriteDao favouriteDao();
}

File: app/src/main/java/ostmodern/skylark/repository/local/SkylarkDatabase.kt
package ostmodern.skylark.repository.local

import android.arch.persistence.room.Database
import android.arch.persistence.room.RoomDatabase

@Database(entities = [(SetEntity::class), (EpisodeEntity::class), (FavouriteEntity::class)],
        version = 1,
        exportSchema = false)
abstract class SkylarkDatabase : RoomDatabase() {
    abstract fun setDao(): SetDao

    abstract fun episodeDao(): EpisodeDao

    abstract fun favouriteDao(): FavouriteDao
}


